Microsoft.Toolkit.Uwp.UI.Controls/QuickReturnHeader/QuickReturnHeader.cs

```diff
@@ -0,0 +1,186 @@
+        /// <summary>
+        /// Gets or sets the ListView this header belongs to
+        /// </summary>
+        public ListView TargetListView { get; set; }
+
+        protected override void OnApplyTemplate()
+        {
+            if (TargetListView != null)
+            {
+                scrollViewer = GetScrollViewer(TargetListView);
+
+                // Place items below header
+                var panel = TargetListView.ItemsPanelRoot;
+                Canvas.SetZIndex(panel, -1);
+            }
+
+            if (scrollViewer != null)
+            {
+                scrollViewer.ViewChanged += (sender, args) =>
+                {
+                    if (animationProperties != null)
+                    {
+                        float oldOffsetY = 0.0f;
+                        animationProperties.TryGetScalar("OffsetY", out oldOffsetY);
+
+                        var delta = scrollViewer.VerticalOffset - previousVerticalScrollOffset;
+                        previousVerticalScrollOffset = scrollViewer.VerticalOffset;
+
+                        var newOffsetY = oldOffsetY - (float)delta;
+
+                        // Keep values within negativ header size and 0
+                        FrameworkElement header = (FrameworkElement)TargetListView.Header;
+                        newOffsetY = Math.Max((float)-header.ActualHeight, newOffsetY);
+                        newOffsetY = Math.Min(0, newOffsetY);
+
+                        if (oldOffsetY != newOffsetY)
+                        {
+                            animationProperties.InsertScalar("OffsetY", newOffsetY);
+                        }
+                    }
+                };
+            }
+
+            SizeChanged += (sender, args) =>
+            {
+                if (TargetListView != null)
+                {
+                    if (IsQuickReturnEnabled)
+                    {
+                        StartAnimation();
+                    }
+                }
+            };
+        }
+
+        private static ScrollViewer GetScrollViewer(DependencyObject o)
+        {
+            if (o is ScrollViewer)
+            {
+                return o as ScrollViewer;
+            }
+
+            for (int i = 0; i < VisualTreeHelper.GetChildrenCount(o); i++)
+            {
+                var child = VisualTreeHelper.GetChild(o, i);
+
+                var result = GetScrollViewer(child);
+                if (result != null)
+                {
+                    return result;
+                }
+            }
+
+            return null;
+        }
+
+        private static void OnIsQuickReturnEnabledChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var me = d as QuickReturnHeader;
+
+            if (me.IsQuickReturnEnabled)
+            {
+                me.StartAnimation();
+            }
+            else
+            {
+                me.StopAnimation();
+            }
+        }
+
+        private void StartAnimation()
```

Comment:
I don't think so. The animation used is an expression based animation that automatically adjusts the offfset in relation to scrolling. The Offset() extension helps with key frame animations or one time property changes if the animation duration is 0.